<h1>Activity 1 - PALMER PENGUINS DATA ANALYITCS PROJECT</h1>
<hr>
<h3> Analyst : Jerwin Nico H. Quijano</h3>

In [1]:
# import libraries
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

<h1>Converting csv into a Data Frame</h1>

<h3>Requirements</h3>
<ul>
    <li>Data Frame name</li>
    <li>CSV File path</li>
</ul>

In [2]:
penguins = pd.read_csv('datasets//penguins.csv')
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h1>Describing Data</h1>
<hr>

In [3]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

In [4]:
penguins.describe().T # transpose

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

<h1>Checking Null values</h1>
<hr>

In [5]:
penguins.isnull().sum() # check the null and sum the null values.

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h1>Checking Values Counts</h1>
<hr>

In [6]:
penguins['species']

0         Adelie
1         Adelie
2         Adelie
3         Adelie
4         Adelie
         ...    
339    Chinstrap
340    Chinstrap
341    Chinstrap
342    Chinstrap
343    Chinstrap
Name: species, Length: 344, dtype: object

In [7]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

<h1>Filtering Data</h1>
<hr>

In [8]:
adelie_DF = penguins[penguins['species'] == "Adelie"]
adelie_DF

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[152 rows x 8 columns]

In [9]:
gentoo_DF = penguins[penguins['species'] == "Gentoo"]
gentoo_DF

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN            NaN                NaN   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 8 columns]

In [10]:
chinstrap_DF = penguins[penguins['species'] == "Chinstrap"]
chinstrap_DF

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

<h1>Questions</h1>
<hr>
<ul>
    <li>How many of these Male Chinstrap species weight is abouve 4000 grams</li>
</ul>

In [11]:
heavy_chinstrap = penguins[(penguins['species'] == 'Chinstrap') & (penguins['body_mass_g'] >= 4000)]

<h1>Fillin up Null Values</h1>
<hr>

In [12]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [13]:
average_bill_len = penguins['bill_length_mm'].mean()
penguins['bill_length_mm'] = penguins['bill_length_mm'].fillna(average_bill_len)
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [14]:
average_bill_depth = penguins['bill_depth_mm'].mean()
penguins['bill_depth_mm'] = penguins['bill_depth_mm'].fillna(average_bill_depth)
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [15]:
average_flipper_ave = penguins['flipper_length_mm'].mean()
penguins['flipper_length_mm'] = penguins['flipper_length_mm'].fillna(average_flipper_ave)
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     0
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [16]:
average_body_mass = penguins['body_mass_g'].mean()

In [17]:
penguins['body_mass_g'] = penguins['body_mass_g'].fillna(average_body_mass)
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     0
body_mass_g           0
sex                  11
year                  0
dtype: int64

In [18]:
frequency_sex = penguins['sex'].mode()
penguins['sex'] = penguins['sex'].fillna(frequency_sex[0])
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

In [19]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     344 non-null    float64
 3   bill_depth_mm      344 non-null    float64
 4   flipper_length_mm  344 non-null    float64
 5   body_mass_g        344 non-null    float64
 6   sex                344 non-null    object 
 7   year               344 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 21.6+ KB


In [20]:
penguins['species'] = penguins['species'].astype('category')
penguins['island'] = penguins['island'].astype('category') 
penguins['sex'] = penguins['sex'].astype('category')
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    int64   
dtypes: category(3), float64(4), int64(1)
memory usage: 15.0 KB


In [21]:
penguins['mass_kg'] = np.round(penguins['body_mass_g'] / 1000, 2)

<h1>Exploratory Data Analysis</h1>
<ol>
    <li>What is the average body mass of each penguin species? Chinstrap: 3733.08 g, Gentoo: 5068.96 g, Adelie: 3703.95 g.</li>
    <li>Which sex in the Gentoo species has the longest average bill length? Male: 59.6 mm, Female: 50.5 mm.</li>
    <li>In what year was the heaviest penguin ever recorded? A male Gentoo penguin from Biscoe Island 6300.0 on the year 2007.</li>
    <li>What is the average flipper length of male penguins on Torgersen Island? 194.03 mm.</li>
    <li>What is the average body mass of Chinstrap penguins per year? 3694.23 g in 2007, 3800.0 g in 2008, and 3725.0 g in 2009.</li>
    <li>What is the total spiecies of adelie penguin in the following year? and what is the Heaviest penguins in the said year?</li>
    <li>What is the most common species of penguin found on each island? Biscoe: Gentoo, Dream: Chinstrap, Torgersen: Adelie.</li>
</ol>


In [22]:
# chinstrap average body mass gram
chinstrap_df = penguins[penguins['species'] == 'Chinstrap']
chinstrap_average_BMG = chinstrap_df['body_mass_g'].mean()
chinstrap_average_BMG

np.float64(3733.0882352941176)

In [23]:
# gentoo average body mass gram
gentoo_df = penguins[penguins['species'] == 'Gentoo']
gentoo_average_BMG = gentoo_df['body_mass_g'].mean()
gentoo_average_BMG

np.float64(5068.965761177136)

In [24]:
# gentoo average body mass gram
adelie_df = penguins[penguins['species'] == 'Adelie']
adelie_average_BMG = adelie_df['body_mass_g'].mean()
adelie_average_BMG

np.float64(3703.95891043398)

In [25]:
# gentoo longest bill lenght in both genders
gentoo_df = penguins[penguins['species'] == 'Gentoo']
gentoo_male_longest_bill_len = gentoo_df[gentoo_df['sex'] == 'male']['bill_length_mm'].max()
gentoo_female_longest_bill_len = gentoo_df[gentoo_df['sex'] == 'female']['bill_length_mm'].max()
print("Male:",gentoo_male_longest_bill_len)
print("Female:",gentoo_female_longest_bill_len)


Male: 59.6
Female: 50.5


<h1>Insight</h1>
<hr>
<p>I found out that in the gentoo's spieces the longest bill lenght in male categorty is 59.6 while on female the longest bill lenght is 50.5. So, therefore that longest bill lengt in Gentoo spieces is Male.</p>

In [26]:
# what year has the heaviest penguin?
max_bmg = penguins['body_mass_g'].max()
print(max_bmg)
penguins[(penguins['year']) & (penguins['body_mass_g'] >= max_bmg)]

6300.0


species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
169  Gentoo  Biscoe            49.2           15.2              221.0   

     body_mass_g   sex  year  mass_kg  
169       6300.0  male  2007      6.3

<h1>Insight</h1>
<hr>
<p>So in the year 2007 the heaviest penguins has ever recorded with the body mass per gram of 63000.</p>

In [27]:
# What is the average flipper lenght of Male penguin in Torgersen island?
penguins[(penguins['sex'] == 'male') & (penguins['island'] == 'Torgersen')]['flipper_length_mm'].mean()

np.float64(194.03268588137007)

<h1>Insight</h1>
<hr>
<p>So it turns out that the average flipper lenght is 194.03 in the Torgersen island</p>

In [28]:
# What is the average body mass of a chinstrap penguin per year? 987
bmg_2007 = penguins[(penguins['species'] == 'Chinstrap') & (penguins['year'] == 2007)]['body_mass_g'].mean()
bmg_2008 = penguins[(penguins['species'] == 'Chinstrap') & (penguins['year'] == 2008)]['body_mass_g'].mean()
bmg_2009 = penguins[(penguins['species'] == 'Chinstrap') & (penguins['year'] == 2009)]['body_mass_g'].mean()
print(bmg_2007)
print(bmg_2008)
print(bmg_2009)


3694.230769230769
3800.0
3725.0


<h1>Insight</h1
<hr>
<p>So the average body mass of a chin strap per year is: 3694.23 in the year 2007, 3800.0 in the year 2008, and 3725.0 in the year 2009.</p>

In [29]:
# What is the total spiecies of adlie penguin in the following year?
adelie_df = penguins[penguins['species'] == 'Adelie']

heaviest_2007 = adelie_df[adelie_df['year'] == 2007]['mass_kg'].max()
heaviest_2007_col = adelie_df[adelie_df['mass_kg'] == heaviest_2007]

heaviest_2008 = adelie_df[adelie_df['year'] == 2008]['mass_kg'].max()
heaviest_2008_col = adelie_df[adelie_df['mass_kg'] == heaviest_2008]

heaviest_2009 = adelie_df[adelie_df['year'] == 2009]['mass_kg'].max()
heaviest_2009_col = adelie_df[adelie_df['mass_kg'] == heaviest_2009]

print(heaviest_2007_col, " ", heaviest_2008_col, " ", heaviest_2009_col)

  species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
7  Adelie  Torgersen            39.2           19.6              195.0   

   body_mass_g   sex  year  mass_kg  
7       4675.0  male  2007     4.68        species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
81  Adelie  Torgersen            42.9           17.6              196.0   

    body_mass_g   sex  year  mass_kg  
81       4700.0  male  2008      4.7         species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
109  Adelie  Biscoe            43.2           19.0              197.0   

     body_mass_g   sex  year  mass_kg  
109       4775.0  male  2009     4.78  


<h1>Insight</h1>
<hr>
<p>So it turns out that in the year 2009 the heaviest penguins is born while in the year 2008 the penguin with the lowest body mass is born.</p>

In [30]:
# What is the most common species of penguin found on each island?
biscoe_island = penguins[penguins['island'] == 'Biscoe']['species'].mode()
dream_island = penguins[penguins['island'] == 'Dream']['species'].mode()
torgersen_island = penguins[penguins['island'] == 'Torgersen']['species'].mode()
print("Biscoe island commmon penguin: ",biscoe_island[0])
print("Dream island commmon penguin: ",dream_island[0])
print("Torgersen island commmon penguin: ",torgersen_island[0])

Biscoe island commmon penguin:  Gentoo
Dream island commmon penguin:  Chinstrap
Torgersen island commmon penguin:  Adelie


<h1>Insight</h1>
<hr>
<p>In Biscoe island the common species of penguins is the Gentoo, while on Dream island the common penguin were Chinstrapp, and on the last island which is Torgersen island the common penguins is Adelie.</p>

In [31]:
biscoed = penguins[penguins['island'] == 'Biscoe']
biscoed[biscoed['sex'] == 'male']['mass_kg'].max()

np.float64(6.3)

In [32]:
penguins.head(20)

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0   Adelie  Torgersen        39.10000       18.70000         181.000000   
1   Adelie  Torgersen        39.50000       17.40000         186.000000   
2   Adelie  Torgersen        40.30000       18.00000         195.000000   
3   Adelie  Torgersen        43.92193       17.15117         200.915205   
4   Adelie  Torgersen        36.70000       19.30000         193.000000   
5   Adelie  Torgersen        39.30000       20.60000         190.000000   
6   Adelie  Torgersen        38.90000       17.80000         181.000000   
7   Adelie  Torgersen        39.20000       19.60000         195.000000   
8   Adelie  Torgersen        34.10000       18.10000         193.000000   
9   Adelie  Torgersen        42.00000       20.20000         190.000000   
10  Adelie  Torgersen        37.80000       17.10000         186.000000   
11  Adelie  Torgersen        37.80000       17.30000         180.000000   
12  Adelie  Torgersen        41.10000       17.60000         182.000000   
13  Adelie  Torgersen        38.60000       21.20000         191.000000   
14  Adelie  Torgersen        34.60000       21.10000         198.000000   
15  Adelie  Torgersen        36.60000       17.80000         185.000000   
16  Adelie  Torgersen        38.70000       19.00000         195.000000   
17  Adelie  Torgersen        42.50000       20.70000         197.000000   
18  Adelie  Torgersen        34.40000       18.40000         184.000000   
19  Adelie  Torgersen        46.00000       21.50000         194.000000   

    body_mass_g     sex  year  mass_kg  
0   3750.000000    male  2007     3.75  
1   3800.000000  female  2007     3.80  
2   3250.000000  female  2007     3.25  
3   4201.754386    male  2007     4.20  
4   3450.000000  female  2007     3.45  
5   3650.000000    male  2007     3.65  
6   3625.000000  female  2007     3.62  
7   4675.000000    male  2007     4.68  
8   3475.000000    male  2007     3.48  
9   4250.000000    male  2007     4.25  
10  3300.000000    male  2007     3.30  
11  3700.000000    male  2007     3.70  
12  3200.000000  female  2007     3.20  
13  3800.000000    male  2007     3.80  
14  4400.000000    male  2007     4.40  
15  3700.000000  female  2007     3.70  
16  3450.000000  female  2007     3.45  
17  4500.000000    male  2007     4.50  
18  3325.000000  female  2007     3.32  
19  4200.000000    male  2007     4.20

In [33]:
penguins.tail(20)

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
324  Chinstrap  Dream            51.5           18.7              187.0   
325  Chinstrap  Dream            49.8           17.3              198.0   
326  Chinstrap  Dream            48.1           16.4              199.0   
327  Chinstrap  Dream            51.4           19.0              201.0   
328  Chinstrap  Dream            45.7           17.3              193.0   
329  Chinstrap  Dream            50.7           19.7              203.0   
330  Chinstrap  Dream            42.5           17.3              187.0   
331  Chinstrap  Dream            52.2           18.8              197.0   
332  Chinstrap  Dream            45.2           16.6              191.0   
333  Chinstrap  Dream            49.3           19.9              203.0   
334  Chinstrap  Dream            50.2           18.8              202.0   
335  Chinstrap  Dream            45.6           19.4              194.0   
336  Chinstrap  Dream            51.9           19.5              206.0   
337  Chinstrap  Dream            46.8           16.5              189.0   
338  Chinstrap  Dream            45.7           17.0              195.0   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  mass_kg  
324       3250.0    male  2009     3.25  
325       3675.0  female  2009     3.68  
326       3325.0  female  2009     3.32  
327       3950.0    male  2009     3.95  
328       3600.0  female  2009     3.60  
329       4050.0    male  2009     4.05  
330       3350.0  female  2009     3.35  
331       3450.0    male  2009     3.45  
332       3250.0  female  2009     3.25  
333       4050.0    male  2009     4.05  
334       3800.0    male  2009     3.80  
335       3525.0  female  2009     3.52  
336       3950.0    male  2009     3.95  
337       3650.0  female  2009     3.65  
338       3650.0  female  2009     3.65  
339       4000.0    male  2009     4.00  
340       3400.0  female  2009     3.40  
341       3775.0    male  2009     3.78  
342       4100.0    male  2009     4.10  
343       3775.0  female  2009     3.78

<h1>Data Aggregation</h1>
<hr>
<ul>
    <li>ilan penguin per island?</li>
    <li>average bill length of a penguin per species and sex.</li>
    <li>What is the maximum body mass of a female penguin in the most island with abundant food?</li>
</ul>

per_island_penguins = penguins.groupby(['island', 'species', 'year']).size()
per_island_penguins

In [34]:
per_species_bill = penguins.groupby(['species', 'sex'])['bill_length_mm'].mean()
per_species_bill

species    sex   
Adelie     female    37.257534
           male      40.273695
Chinstrap  female    46.573529
           male      51.094118
Gentoo     female    45.563793
           male      49.156393
Name: bill_length_mm, dtype: float64

<h1>Correlation Analysis</h1>
<hr>
<h2>Testing Correlation using Pearson R</h2>
<hr>
<p>Pearson R can test correlation between two Variable, given that both variables are <b>NUMERICAL DATA</b></p>
<p>Pearson R only use for both <b>NUMERICAL DATA</b></p>
<p>p - value</p>

<p>if value is >= 0.5 <b>CORRELATED</b></p>
<p>if value is <= 0.5 <b>NOT CORRELATED</b></p>

In [35]:
corr_body_mass = penguins['']

KeyError: ''

In [ ]:
# import scipy.stats
from scipy.stats import f_oneway

In [ ]:
# filter male body mass
penguins[penguins['sex'] == 'male']['body_mass_g']

In [ ]:
# filter female body mass
penguins[penguins['sex'] == 'female']['body_mass_g']

In [ ]:
stats, p_value = f_oneway(penguins[penguins['sex'] == 'male']['body_mass_g'], penguins[penguins['sex'] == 'female']['body_mass_g'])
p_value